kv/union_store.go

```diff
@@ -14,18 +14,17 @@
 package kv
 
 import (
-	"bytes"
-
 	"github.com/pingcap/errors"
 )
 
 // UnionStore is a store that wraps a snapshot for read and a BufferStore for buffered write.
 // Also, it provides some transaction related utilities.
 type UnionStore interface {
 	MemBuffer
-	// CheckLazyConditionPairs loads all lazy values from store then checks if all values are matched.
-	// Lazy condition pairs should be checked before transaction commit.
-	CheckLazyConditionPairs() error
+	// Returns true if k should not exist before.
+	ShouldNotExist(k Key) bool
```

Comment:
`ShouldNotExist` and `LookupConditionErr` are nearly the same, should we merge them into one function ?

Reply:
Good idea.

Reply:
Done